pkg/chains/storage/ca/ca.go

```diff
@@ -0,0 +1,350 @@
+		// get payload identifier
+		name := occ.GetResource().GetUri()
+		// get "Payload" field from the occurrence
+		payload := occ.GetAttestation().GetAttestation().GetGenericSignedAttestation().GetSerializedPayload()
+
+		result[name] = string(payload)
+	}
+
+	return result, nil
+}
+
+// Retrieve signatures from container analysis and store it in a map
+func (b *Backend) RetrieveSignatures(opts config.StorageOpts) (map[string][]string, error) {
+	// initialize an empty map for result
+	result := make(map[string][]string)
+
+	// get all occurrences created using this backend
+	occurrences, err := b.getOccurrences()
+	if err != nil {
+		return nil, err
+	}
+
+	for _, occ := range occurrences {
+		// get the Signature identifier
+		name := occ.GetResource().GetUri()
+		// get "Signatures" field from the occurrence DSSE envelop
+		// signatures := occurrence.GetEnvelope().GetSignatures()
+		signatures := occ.GetAttestation().GetAttestation().GetGenericSignedAttestation().Signatures
+		// unmarshal signatures
+		unmarshalSigs := []string{}
+		for _, sig := range signatures {
+			unmarshalSigs = append(unmarshalSigs, string(sig.GetSignature()))
+		}
+
+		result[name] = unmarshalSigs
+	}
+
+	return result, nil
+}
+
+func (b *Backend) Type() string {
+	return StorageBackendCA
+}
+
+// ----------------------------- Helper Functions ----------------------------
+func (b *Backend) createNote() error {
+	noteID := b.cfg.Storage.ContainerAnalysis.NoteID
+
+	b.logger.Infof("Creating a note with note name %s", noteID)
+
+	// create note request
+	noteReq := &pb.CreateNoteRequest{
+		Parent: b.getProjectPath(),
+		NoteId: noteID,
+		Note: &pb.Note{
+			// Name:             b.getNotePath(),
+			ShortDescription: "An attestation note",
+			Kind:             commonpb.NoteKind_ATTESTATION,
+			Type: &pb.Note_AttestationAuthority{
+				AttestationAuthority: &attestationpb.Authority{
+					Hint: &attestationpb.Authority_Hint{
+						HumanReadableName: "This note was auto-generated by Tekton Chains",
+					},
+				},
+			},
+		},
+	}
+
+	// store note
+	_, err := b.client.CreateNote(context.Background(), noteReq)
+
+	if err != nil {
+		// noteID already exisits
+		return err
+	}
+	return nil
+}
+
+func (b *Backend) createOccurrenceRequest(payload []byte, signature string, opts config.StorageOpts) *pb.CreateOccurrenceRequest {
+	occurrenceDetails := &pb.Occurrence_Attestation{
+		Attestation: &attestationpb.Details{
+			Attestation: &attestationpb.Attestation{
+				Signature: &attestationpb.Attestation_GenericSignedAttestation{
+					GenericSignedAttestation: &attestationpb.GenericSignedAttestation{
+						// Uspecified ContentType because we have simplesigning for OCI, In-toto for TaskRun
+						ContentType:       attestationpb.GenericSignedAttestation_CONTENT_TYPE_UNSPECIFIED,
+						SerializedPayload: payload,
+						Signatures: []*commonpb.Signature{
+							{
+								Signature:   []byte(signature),
+								PublicKeyId: b.cfg.Signers.KMS.KMSRef,
```

Comment:
What happens if a KMS key isn't used (e.g. k8s secret, fulcio)?

Reply:
I'll add todo comment here stating that we will add other key refs later.